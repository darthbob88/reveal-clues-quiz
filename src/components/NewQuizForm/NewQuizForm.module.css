.question {
  margin-top: 1rem;
  padding: 0.5rem;
}
ul.clues {
  max-width: 60%;
  margin: 0 auto;
  list-style-position: inside;
}
.question button {
  display: block;
  margin: 0.5rem auto;
}
.question input {
  margin-left: 0.5rem;
}
span.prompt {
  font-size: 1.1em;
  font-weight: bolder;
}
