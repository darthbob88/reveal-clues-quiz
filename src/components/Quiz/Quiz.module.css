.question {
  border: 1px solid #bbb;
  margin-top: 1rem;
  padding: 0.5rem;
}
.disabled {
  background: #bcbcbc;
  color: #222;
}
.quiz {
  border: 1px solid lightgray;
  padding-bottom: 1em;
}
.startQuiz {
  opacity: 100%;
  display: inline-block;
  margin: 2rem 0.5rem 4rem;
}

div.nextPrevQs {
  display: flex;
  justify-content: center;
}
div.nextPrevQs button {
  margin: 0 1rem;
}
.questions li {
  display: inline-block;
  padding: 0.3rem;
  margin: 0.2rem;
  cursor: pointer;
}

.unanswered {
  background-color: #e6e6fa4f;
}
.correct {
  background-color: #7cfc007a;
}
.incorrect {
  background-color: #ff0000a6;
  color: white;
}

.current {
  border: 1px solid black;
}
