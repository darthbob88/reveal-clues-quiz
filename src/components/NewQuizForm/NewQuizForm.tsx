import React, { useState } from "react";
import { Question } from "../../model/Question";
import { emptyQuestion, emptyQuiz, Quiz } from "../../model/Quiz";
import { saveNewQuiz } from "../../model/QuizService";
import styles from "./NewQuizForm.module.css";

/* TODO: Remaining tasks
Add an authorship field; this may require integrating with Firebase to get user ID.
Add a field for time. Either a dropdown in 30s intervals, or a number input for number of minutes.
Add some upper limits; no more than 10 questions and 10 clues each, or whatever.
Validation; require all fields, at least one non-blank clue and answer for each question, etc.
*/
export const NewQuizForm: React.FunctionComponent = () => {
  const blankQuiz: Quiz = { ...emptyQuiz, questions: [{ clues: ["", "", "", ""], answer: "", revealOnAnswer: "" }] };
  const [newQuiz, setNewQuiz] = useState(blankQuiz);

  const canRemoveQuestion = newQuiz.questions.length > 1;
  const removeQuestion = (index: number) => {
    let tempQs = newQuiz.questions;
    if (newQuiz.questions.length <= 1)
      return;
    tempQs.splice(index, 1);
    setNewQuiz({ ...newQuiz, questions: tempQs });
  };

  const addQuestion = () => {
    let tempQs = newQuiz.questions;
    tempQs.push(emptyQuestion);
    setNewQuiz({ ...newQuiz, questions: tempQs });
  };

  const updateQuestion = (QIdx: number, updatedQuestion: Question) => {
    let tempQs = newQuiz.questions;
    tempQs[QIdx] = updatedQuestion;
    setNewQuiz({ ...newQuiz, questions: tempQs });
  }

  const submitQuiz = () => {
    // TODO Fix this slugification
    newQuiz.slug = newQuiz.title.toLocaleLowerCase().replace(/\s/g, "-");
    saveNewQuiz(newQuiz);
  };

  return (
    <form className={styles.question}>
      <p>
        <label>
          Quiz Title
          <input
            type="text"
            onChange={(evt) =>
              setNewQuiz({ ...newQuiz, title: evt.target.value })
            }
            value={newQuiz.title}
          />
        </label>
      </p>
      <p>
        <label>
          Quiz Prompt
          <input
            type="text"
            onChange={(evt) =>
              setNewQuiz({ ...newQuiz, prompt: evt.target.value })
            }
            value={newQuiz.prompt}
          />{" "}
        </label>
      </p>
      <button
        onClick={(event) => {
          event.preventDefault();
          addQuestion();
        }}
      >
        Add Question
      </button>
      {newQuiz.questions.map((question, QIdx) => {
        return (
          <SingleQuestion key={QIdx} question={question} QIdx={QIdx} removeQuestion={removeQuestion} canRemoveQuestion={canRemoveQuestion} updateQuestion={updateQuestion} />);
      })}
      <button
        onClick={(event) => {
          event.preventDefault();
          submitQuiz();
        }}
      >
        Submit New Quiz
      </button>
    </form>
  );
};

type SingleQuestionProps = {
  question: Question;
  QIdx: number;
  updateQuestion: Function;
  removeQuestion: Function;
  canRemoveQuestion: boolean
}

const SingleQuestion: React.FunctionComponent<SingleQuestionProps> = ({ question, QIdx, updateQuestion, removeQuestion, canRemoveQuestion }) => {

  const updateClue = (event: React.ChangeEvent<HTMLInputElement>, clueIdx: number) => {
    const newClue = event.target.value;
    const updatedQuestion = question;
    updatedQuestion.clues[clueIdx] = newClue;
    updateQuestion(QIdx, updatedQuestion);
  }

  const updateAnswer = (event: React.ChangeEvent<HTMLInputElement>) => {
    const newAnswer = event.target.value;
    const updatedQuestion = question;
    updatedQuestion.answer = newAnswer;
    updateQuestion(QIdx, updatedQuestion);
  }
  const updateAddedText = (event: React.ChangeEvent<HTMLInputElement>) => {
    const newAnswer = event.target.value;
    const updatedQuestion = question;
    updatedQuestion.revealOnAnswer = newAnswer;
    updateQuestion(QIdx, updatedQuestion);
  }

  const addClue = () => {
    const updatedQuestion = question;
    updatedQuestion.clues.push("");
    updateQuestion(QIdx, updatedQuestion);
  }
  const removeClue = (clueIdx: number) => {
    let tempClues = question.clues;
    tempClues.splice(clueIdx, 1);

    updateQuestion(QIdx, { ...question, clues: tempClues });
  }

  return (<div data-testid={"question" + (QIdx + 1)}>
    <h3>Question {QIdx + 1}</h3>
    <button data-testid="remove-question"
      disabled={!canRemoveQuestion}
      onClick={(event) => {
        event.preventDefault();
        removeQuestion(QIdx);
      }}
    >
      X
    </button>
    <button data-testid="add-clue"
      onClick={(event) => {
        event.preventDefault();
        addClue();
      }}
    >
      Add Clue
    </button>
    <ol>
      {question.clues.map((clue, clueIdx) => (
        <li key={clueIdx} >
          <input type="text" data-index={clueIdx} data-testid={"clue" + (clueIdx + 1)} value={clue} onChange={(event) => updateClue(event, clueIdx)} />
          <button data-testid="remove-clue" disabled={question.clues.length <= 1}
            onClick={(event) => {
              event.preventDefault();
              removeClue(clueIdx);
            }}
          >
            X
          </button>
        </li>
      ))}
    </ol>
    <label>
      Answer <input type="text" onChange={(event) => updateAnswer(event)} value={question.answer} />
    </label>
    <label>
      Additional text <input type="text" onChange={(event) => updateAddedText(event)} value={question.revealOnAnswer} />
    </label>
  </div>)

}