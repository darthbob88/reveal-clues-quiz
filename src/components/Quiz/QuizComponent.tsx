import React, { useState } from "react";
import { QuizState } from "../../model/Quiz";
import styles from "./Quiz.module.css";
import { QuestionComponent } from "../Question/Question";
import { QuestionEnum, QuestionState } from "../../model/Question";
import { observer } from "mobx-react";

type QuizProps = {
  quiz: QuizState;
};
export const QuizComp: React.FunctionComponent<QuizProps> = ({ quiz }) => {
  const [currentQuestion, setCurrentQuestion] = useState(0);
  const questions = quiz.currentQuiz.questions;

  const prevQuestion = () => {
    // This is stupid, but I can't think of a much better method
    let prev = questions.length + currentQuestion - 1;
    while (
      quiz.quizState[prev % questions.length].state !== QuestionEnum.UNANSWERED
    ) {
      prev--;
    }
    setCurrentQuestion(prev % questions.length);
  };

  const nextQuestion = () => {
    let next = currentQuestion + 1;
    while (
      quiz.quizState[next % questions.length].state !== QuestionEnum.UNANSWERED
    ) {
      next++;
    }
    setCurrentQuestion(next % questions.length);
  };

  const scoreQuestion = (points: QuestionState) => {
    quiz.scoreQuestion(currentQuestion, points);
    setTimeout(() => {
      nextQuestion();
    }, 800);
  };

  return (
    <div className={styles.quiz}>
      <p>
        Time remaining: {quiz.display} <br />
        Current score: {quiz.scorePoints} pts {quiz.scorePercent}/
        {quiz.quizState.length} question(s) correct
        {quiz.startedQuiz ? "" : <button className={styles.startQuiz} onClick={quiz.startQuiz}>Start Quiz</button>}
      </p>
      <div className={`${!quiz.startedQuiz ? styles.disabled : ""}`}>
        <ul className={styles.questions}>
          {quiz.quizState.map((question, index) => (
            <li
              key={index}
              onClick={() => setCurrentQuestion(index)}
              className={`${index === currentQuestion ? styles.current : ""}
              ${
                //TODO: This is ugly, but I can't find a good way to make this a mapping.
                question.state === QuestionEnum.UNANSWERED
                  ? styles.unanswered
                  : question.state === QuestionEnum.CORRECTLY_ANSWERED
                    ? styles.correct
                    : styles.incorrect
                }`}
            >
              {index + 1}
            </li>
          ))}
        </ul>
        <button onClick={() => prevQuestion()}>&lt; Previous Question</button>
        <button onClick={() => nextQuestion()}>Next Question &gt;</button>
        <QuestionComponent
          question={questions[currentQuestion]}
          state={quiz.quizState[currentQuestion]}
          awardPoints={scoreQuestion}
        />
      </div>
    </div>
  );
};

export const QuizComponent = observer(QuizComp);
