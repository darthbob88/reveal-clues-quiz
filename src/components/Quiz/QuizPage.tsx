import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { Quiz, QuizState } from "../../model/Quiz";
import { loadQuiz } from "../../model/QuizService";
import { QuizComponent } from "./QuizComponent";

export const QuizPage: React.FunctionComponent = () => {
    const { id } = useParams<{ id: string }>();
    const [quiz, setQuiz] = useState<Quiz>();
    useEffect(() => {
        const fetchData = async (id: string) => {
            const result = await loadQuiz(id);
            setQuiz(result);
        };
        id != null && fetchData(id);
    }, [id]);
    return (
        quiz === undefined ? <h2>Still Loading</h2> : <QuizComponent quiz={new QuizState(quiz)} />
    );
}