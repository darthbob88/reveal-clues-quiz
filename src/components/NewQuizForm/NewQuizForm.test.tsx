import React from "react";
import { fireEvent, render, within } from "@testing-library/react";
import { NewQuizForm } from "./NewQuizForm";
import { testQuizzes } from "../../model/Quiz";
import userEvent from "@testing-library/user-event/";
import * as QuizService from "../../model/QuizService";

describe("New Quiz component", () => {
  test("renders the new quiz form properly", () => {
    const { container } = render(
      <NewQuizForm />
    );

    expect(container).toMatchSnapshot();
  });
  describe("Properly handles adding/removing clues/questions", () => {
    test("adds a new question when the button is clicked", () => {
      const { getByText, queryByText } = render(
        <NewQuizForm />
      );

      let result = queryByText(/Question 2/i);
      expect(result).not.toBeInTheDocument();

      const addQBtn = getByText(/Add Question/i);
      fireEvent.click(addQBtn);

      result = queryByText(/Question 2/i);
      expect(result).toBeInTheDocument();
    });

    test("Can only add up to 10 questions", () => {
      const { getByText, getByTestId } = render(
        <NewQuizForm />
      );

      const addQBtn = getByText(/Add Question/i);
      for (let ii = 0; ii < 10; ii++) {
        fireEvent.click(addQBtn);
        expect(getByTestId(`question${ii + 1}`)).toBeInTheDocument();
      }
      expect(addQBtn).toBeDisabled();
    });

    test("removes a question when the button is clicked", () => {
      const { getByText, getByTestId, queryByTestId } = render(
        <NewQuizForm />
      );

      const addQBtn = getByText(/Add Question/i);
      fireEvent.click(addQBtn);

      const question = getByTestId("question2");
      expect(question).toBeInTheDocument();

      const removeQBtn = within(question).getByTestId("remove-question");
      fireEvent.click(removeQBtn);

      const question2 = queryByTestId("question2");
      expect(question2).not.toBeInTheDocument();
    });

    test("Cannot remove question if only 1 left", () => {
      const { getByTestId } = render(
        <NewQuizForm />
      );

      const question = getByTestId("question1");
      expect(question).toBeInTheDocument();

      const removeQBtn = within(question).getByTestId("remove-question");
      expect(removeQBtn).toBeDisabled();
    });

    test("adds a clue to a question when the button is clicked", () => {
      const { getByTestId } = render(
        <NewQuizForm />
      );

      const question = getByTestId("question1");
      expect(question).toBeInTheDocument();

      const clue1 = within(question).queryByTestId("clue1");
      expect(clue1).toBeInTheDocument();

      let clue5 = within(question).queryByTestId("clue5");
      expect(clue5).not.toBeInTheDocument();

      const addClueQBtn = within(question).getByTestId("add-clue");
      fireEvent.click(addClueQBtn);

      clue5 = within(question).queryByTestId("clue5");
      expect(clue5).toBeInTheDocument();
    });

    test("Can only add up to 10 clues", () => {
      const { getByTestId } = render(
        <NewQuizForm />
      );

      const question = getByTestId("question1");
      expect(question).toBeInTheDocument();

      const addClueQBtn = within(question).getByTestId("add-clue");

      for (let ii = 4; ii < 10; ii++) {
        fireEvent.click(addClueQBtn);
        expect(getByTestId(`clue${ii + 1}`)).toBeInTheDocument();
      }
      expect(addClueQBtn).toBeDisabled();
    });

    test("removes a clue from a question when the button is clicked", () => {
      const { getByTestId } = render(
        <NewQuizForm />
      );

      const question = getByTestId("question1");
      expect(question).toBeInTheDocument();

      const clue1 = within(question).queryByTestId("clue1");
      expect(clue1).toBeInTheDocument();

      let clue4 = within(question).queryByTestId("clue4");
      expect(clue4).toBeInTheDocument();

      const removeClueQBtn = within(question).getAllByTestId("remove-clue");
      expect(removeClueQBtn).toHaveLength(4);
      fireEvent.click(removeClueQBtn[0]);

      clue4 = within(question).queryByTestId("clue4");
      expect(clue4).not.toBeInTheDocument();
    });


    test("Cannot remove clue if only 1 left", () => {
      const { getByTestId } = render(
        <NewQuizForm />
      );

      const question = getByTestId("question1");
      expect(question).toBeInTheDocument();

      let removeClueQBtns = within(question).getAllByTestId("remove-clue");
      for (let ii = removeClueQBtns.length -1; ii > 0; ii--) {
        fireEvent.click(removeClueQBtns[ii]);
      }

      removeClueQBtns = within(question).getAllByTestId("remove-clue");
      expect(removeClueQBtns).toHaveLength(1);
      expect(removeClueQBtns[0]).toBeDisabled();
    });

  });

  describe("Properly handles input", () => {
    test("Can handle inputting the title", async () => {
      const user = userEvent.setup();
      const testQuiz = testQuizzes[0];
      const { getByLabelText } = render(
        <NewQuizForm />
      );

      const quizTitle = getByLabelText("Quiz Title");
      await user.type(quizTitle, testQuiz.title);
      expect(quizTitle).toHaveValue(testQuiz.title);
    });

    test("Can handle inputting a prompt", async () => {
      const user = userEvent.setup();
      const testQuiz = testQuizzes[0];
      const { getByLabelText } = render(
        <NewQuizForm />
      );

      const quizPrompt = getByLabelText("Quiz Prompt");
      await user.type(quizPrompt, testQuiz.prompt);
      expect(quizPrompt).toHaveValue(testQuiz.prompt);
    });

    test("Can handle inputting clues for a question", async () => {
      const user = userEvent.setup();
      const testQuiz = testQuizzes[0];
      const clues = testQuiz.questions[0].clues;
      const { getAllByTestId } = render(
        <NewQuizForm />
      );

      const cluePrompts = getAllByTestId(/clue\d/);
      for (let ii = 0; ii < cluePrompts.length; ii++) {
        const cluePrompt = cluePrompts[ii];
        const clue = clues[ii];
        await user.type(cluePrompt, clue);
        expect(cluePrompt).toHaveValue(clue);
      }
    });

    test("Can handle inputting answer for a question", async () => {
      const testQuiz = testQuizzes[0];
      const testAnswer = testQuiz.questions[0].answer;
      const { getByLabelText } = render(
        <NewQuizForm />
      );

      const answerInput = getByLabelText("Answer");
      await userEvent.type(answerInput, testAnswer);
      expect(answerInput).toHaveValue(testAnswer);

    });

    test("Can handle inputting additional text for a question", async () => {
      const testReveal = "butts";
      const { getByLabelText } = render(
        <NewQuizForm />
      );

      const answerInput = getByLabelText("Additional text");
      await userEvent.type(answerInput, testReveal);
      expect(answerInput).toHaveValue(testReveal);

    });
  })

  describe("Properly handles submitting a quiz", () => {
    // Input a test quiz and make sure it gets submitted correctly.
    test("Can handle submitting a good quiz with one question", async () => {
      const saveNewQuizSpy = jest.spyOn(QuizService, "saveNewQuiz");

      const user = userEvent.setup();
      const testQuiz = { ...testQuizzes[0] };
      testQuiz.questions = [{ ...testQuiz.questions[0], revealOnAnswer: "butts" }];
      const { getByLabelText, getAllByTestId, getByText } = render(
        <NewQuizForm />
      );

      const quizTitle = getByLabelText("Quiz Title");
      await user.type(quizTitle, testQuiz.title);
      expect(quizTitle).toHaveValue(testQuiz.title);

      const quizPrompt = getByLabelText("Quiz Prompt");
      await userEvent.type(quizPrompt, testQuiz.prompt);
      expect(quizPrompt).toHaveValue(testQuiz.prompt);

      const timePrompt = getByLabelText("Time Limit (in minutes)");
      const timeInMinutes = testQuiz.time / 60;
      await userEvent.type(timePrompt, `${timeInMinutes}`);
      expect(timePrompt).toHaveValue(timeInMinutes);

      const cluePrompts = getAllByTestId(/clue\d/);
      for (let ii = 0; ii < cluePrompts.length; ii++) {
        const cluePrompt = cluePrompts[ii];
        const clue = testQuiz.questions[0].clues[ii];
        await user.type(cluePrompt, clue);
        expect(cluePrompt).toHaveValue(clue);
      }

      const testReveal = "butts";
      const revealInput = getByLabelText("Additional text");
      await userEvent.type(revealInput, testReveal);
      expect(revealInput).toHaveValue(testReveal);

      const testAnswer = testQuiz.questions[0].answer;
      const answerInput = getByLabelText("Answer");
      await userEvent.type(answerInput, testAnswer);
      expect(answerInput).toHaveValue(testAnswer);

      const submitBtn = getByText("Submit New Quiz");
      fireEvent.click(submitBtn);

      expect(saveNewQuizSpy).toBeCalledTimes(1);
      expect(saveNewQuizSpy).toBeCalledWith(testQuiz);

    });

    jest.setTimeout(10_000);
    test("Can handle submitting a good quiz with multiple questions", async () => {
      const saveNewQuizSpy = jest.spyOn(QuizService, "saveNewQuiz");

      const user = userEvent.setup();
      const testQuiz = { ...testQuizzes[0] };
      testQuiz.questions = [{ ...testQuiz.questions[0], revealOnAnswer: "butts" }, { ...testQuiz.questions[1], revealOnAnswer: "foo" }];

      const { getByLabelText, getByTestId, getByText } = render(
        <NewQuizForm />
      );

      const quizTitle = getByLabelText("Quiz Title");
      await user.type(quizTitle, testQuiz.title);
      expect(quizTitle).toHaveValue(testQuiz.title);

      const quizPrompt = getByLabelText("Quiz Prompt");
      await userEvent.type(quizPrompt, testQuiz.prompt);
      expect(quizPrompt).toHaveValue(testQuiz.prompt);

      const timePrompt = getByLabelText("Time Limit (in minutes)");
      const timeInMinutes = testQuiz.time / 60;
      await userEvent.type(timePrompt, `${timeInMinutes}`);
      expect(timePrompt).toHaveValue(timeInMinutes);

      const addQBtn = getByText(/Add Question/i);
      await userEvent.click(addQBtn);

      for (let jj = 0; jj < testQuiz.questions.length; jj++) {
        const questionDiv = getByTestId(`question${jj + 1}`);
        const questionText = testQuiz.questions[jj];

        const cluePrompts = within(questionDiv).getAllByTestId(/clue\d/);
        for (let ii = 0; ii < cluePrompts.length; ii++) {
          const cluePrompt = cluePrompts[ii];
          const clue = questionText.clues[ii];
          await user.type(cluePrompt, clue);
          // expect(cluePrompt).toHaveValue(clue);
        }

        const testReveal = questionText.revealOnAnswer as string;
        const revealInput = within(questionDiv).getByLabelText("Additional text");
        await userEvent.type(revealInput, testReveal);
        // expect(revealInput).toHaveValue(testReveal);

        const testAnswer = questionText.answer;
        const answerInput = within(questionDiv).getByLabelText("Answer");
        await userEvent.type(answerInput, testAnswer);
        // expect(answerInput).toHaveValue(testAnswer);
      }

      const submitBtn = getByText("Submit New Quiz");
      fireEvent.click(submitBtn);

      expect(saveNewQuizSpy).toBeCalledTimes(1);
      expect(saveNewQuizSpy).toBeCalledWith(testQuiz);

    });
  });
});