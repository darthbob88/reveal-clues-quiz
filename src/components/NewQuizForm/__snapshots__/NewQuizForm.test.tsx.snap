// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Quiz component renders the new quiz form properly 1`] = `
<div>
  <form
    class="question"
  >
    <p>
      <label>
        Quiz Title
        <input
          type="text"
          value="States by Attraction"
        />
      </label>
    </p>
    <p>
      <label>
        Quiz Prompt
        <input
          type="text"
          value="in what state will you find..."
        />
         
      </label>
    </p>
    <button>
      Add Question
    </button>
    <div>
      <h3>
        Question 
        1
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="Ernest Hemingway's Grave"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="The Smurf Turf"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="The Yellowstone Zone of Death"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="The <State> Potato Museum"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Idaho"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        2
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="Weeki Wachee, the city of mermaids"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="The Ernest Hemingway Home and Museum"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="NASA Vehicle Assembly Building, a building so large it has its own weather"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="The Southernmost Point in the Continental US"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Florida"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        3
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="World's Largest Collection of Smallest Versions of Largest Things"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="Monument Rocks"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="The Equality House, across the street from Westboro"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="Geographic Center of the Contiguous United States"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Kansas"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        4
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="Mutter Museum of medical oddities"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="Fallingwater, designed by Frank Lloyd Wright"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="Gobbler's Knob, home of Punxsutawney Phil"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="The grave of Fred McFeely Rogers"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Pennsylvania"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        5
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="Slocum Mechanical Puzzle Collection"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="Mentone Egg, A giant concrete egg marks this town as the \\"Egg Basket of the Midwest.\\""
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="Kokomantis"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="Hoosier Hill"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Indiana"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        6
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="USPS Remote Encoding Facility"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="Pando, 'The Trembling Giant', a massive grove of quaking aspen"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="The Spiral Jetty, a sculpture built onto a lake which is only visible during droughts"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="Bonneville Salt Flats/Speedway"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Utah"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        7
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="The Very Large Array radiotelescope, outside of Socorro."
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="Smokey Bear's Grave"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="Carlsbad Caverns"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="The International UFO Museum and Research Center"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="New Mexico"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        8
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="Vacuum Cleaner Museum"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="Vista House at Crown Point"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="Evergreen Aviation Museum, home of the Spruce Goose"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="International Rose Test Garden, in the City of Roses"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Oregon"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        9
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="Rhea County Courthouse, site of the Scopes monkey trial"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="Concrete Parthenon, built for the 1897 Centennial Exposition"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="National Civil Rights Museum, established in the hotel where Dr Martin Luther King Jr was assassinated"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="Jack Daniels Distillery"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Tennessee"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        10
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="Dr. Seuss House aka The Goose Creek Tower, a 185 foot tall log cabin"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="The giant metal chicken statue, outside the town of Chicken"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="St Michael's Cathedral, first Russian Orthodox Church in North America"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="Santa Claus House, in the town of North Pole"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Alaska"
        />
      </label>
    </div>
    <div>
      <h3>
        Question 
        11
      </h3>
<<<<<<< HEAD
<<<<<<< HEAD
=======
       
      <button>
        X
      </button>
       
>>>>>>> WIP form
      <ul>
=======
      <button>
        X
      </button>
      <button>
        Add Clue
      </button>
      <ol>
>>>>>>> Added capability to update questions
        <li>
          <input
            data-index="0"
            type="text"
            value="Ape Cave, longest continuous lava tube in the continental US"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="1"
            type="text"
            value="Maryhill Museum and Stonehenge"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="2"
            type="text"
            value="Brown and Haley factory store, manufacturers of Almond Roca"
          />
          <button>
            X
          </button>
        </li>
        <li>
          <input
            data-index="3"
            type="text"
            value="Beacon Rock, one of the largest monoliths in the world"
          />
          <button>
            X
          </button>
        </li>
      </ol>
      <label>
        Answer 
        <input
          type="text"
          value="Washington"
        />
      </label>
    </div>
    <button>
      Submit New Quiz
    </button>
  </form>
</div>
`;
